learn/pkg/question/question.go

```diff
@@ -189,32 +191,43 @@ func (m *Model) PrintHTML(buf *bytes.Buffer) {
 		block.PrintHTML(buf)
 	}
 	buf.WriteString("</form>\n")
+	return nil
 }
 
 // ToHTML returns a complete standalone HTML document as string.
-func (m *Model) ToHTML() string {
+func (m *Model) ToHTML(withMarked bool) (string, error) {
 	buf := &bytes.Buffer{}
 	buf.WriteString(questionPrefixHTML)
-	m.PrintHTML(buf)
+	if err := m.PrintHTML(buf, withMarked); err != nil {
+		return "", err
+	}
 	buf.WriteString(questionSuffixHTML)
-	return buf.String()
+	return buf.String(), nil
 }
 
 func baseFilename(filename string) string {
 	return strings.TrimSuffix(filepath.Base(filename), filepath.Ext(filename))
 }
 
-func (m *Model) printAnswerChoicesHTML(list *markdown.List, buf *bytes.Buffer) {
+func (m *Model) printAnswerChoicesHTML(list *markdown.List, buf *bytes.Buffer, withMarked bool) error {
 	buf.WriteString("<fieldset>\n")
 	inputType := "radio"
 	if m.Frontmatter.AnswerType == "multiple-choice" {
 		inputType = "checkbox"
 	}
+	checkedByIndex := map[int]string{}
```

Comment:
Rather that building up a map of just indices to a constant string, perhaps consider getting the answer and asking it if if the index is a correct answer:

```
        var answer Answer
        if withMarked {
                var err error
                answer, err = m.Frontmatter.getAnswer(m.privateKey)
                if err != nil {
                        return err
                }
        }
        for i, item := range list.Items {
                checked := ""
                if answer.correctAnswerIndices()[i] {
                        checked = "checked" // note: would put space at end here. this passes current tests though
                }
 ```

Just a suggestion. I spent a little time understanding this `checkedByIndex` map so my thoughts went down this path. Up to you.

Also, this works because the zero value `Answer` has no correct answers so if `answer` is not set, it will never result in a checked input.

Reply:
Thoughts after :shower: 

That `answer.correctAnswerIndices()[i]` each time in the loop is a little silly - it rebuilds that map each time. So perhaps instead of storing the `answer`, store the `map[int]bool` from `correctAnswerIndices()` and look that up instead of what you currently have of generating a new map?

Reply:
ok, i have squashed you change.
Fun fact: I had nearly exactly what you're suggesting at first and removed it feeling there is too much going on in the single function.
thanks for taking the time and posting the commit.